src/cpp/writer/tswriter.cpp

```diff
@@ -10,126 +11,177 @@
 namespace bfiocpp {
 
 TsWriterCPP::TsWriterCPP(
-    const std::string& fname, 
-    const std::vector<std::int64_t>& image_shape, 
+    const std::string& fname,
+    const std::vector<std::int64_t>& image_shape,
     const std::vector<std::int64_t>& chunk_shape,
-    const std::string& dtype_str
-  ): _filename(fname), 
-     _image_shape(image_shape), 
-     _chunk_shape(chunk_shape),  
+    const std::string& dtype_str,
+    const std::string& dimension_order
+  ): _filename(fname),
+     _image_shape(image_shape),
+     _chunk_shape(chunk_shape),
      _dtype_code(GetDataTypeCode(dtype_str)) {
- 
-  TENSORSTORE_CHECK_OK_AND_ASSIGN(_source, tensorstore::Open(
-                            GetZarrSpecToWrite(_filename, _image_shape, _chunk_shape, GetEncodedType(_dtype_code)),
-                            tensorstore::OpenMode::create |
-                            tensorstore::OpenMode::delete_existing,
-                            tensorstore::ReadWriteMode::write).result());
-}
 
+    
+    TENSORSTORE_CHECK_OK_AND_ASSIGN(_source, tensorstore::Open(
+        GetZarrSpecToWrite(_filename, _image_shape, _chunk_shape, GetEncodedType(_dtype_code)),
+        tensorstore::OpenMode::create |
+        tensorstore::OpenMode::delete_existing,
+        tensorstore::ReadWriteMode::write).result()
+    );
 
-void TsWriterCPP::WriteImageData(py::array& py_image) {
+    auto position = dimension_order.find("X");
```

Comment:
Since "YX" will always be in `dimension_order`, this check can be improved. 
Instead of this, check if `dimenstion_order` is of length between 2 and 5 and ends with `YX`. If not, throw an exception. Also, `_x_index`  is always `0` and '_y_index` is always `1`. These don't need to be member variable.